chsdi/templates/luftbilder/viewer.mako

```diff
@@ -260,7 +279,58 @@
         view.on('propertychange', debounce(updateUrl));
         updateUrl();
 
+        tile.on('postcompose', function(event) {
+          filter(event.context, lubisMap);
+        });
+
+        function filter(context, lubisMap) {
+          var canvas = context.canvas;
+          var brightnessSlider = document.getElementById("brightness");
+          var contrastSlider = document.getElementById("contrast");
+          var saturationSlider = document.getElementById("saturation");
+          brightnessSlider.addEventListener("change", brightnessFunction(canvas, brightnessSlider.value, lubisMap));
+          contrastSlider.addEventListener("change", contrastFunction(canvas, contrastSlider.value, lubisMap));
+          saturationSlider.addEventListener("change", saturationFunction(canvas, saturationSlider.value, lubisMap));
```

Comment:
`addEventListener` doesn't work on IE

Reply:
On change of slider's value you just need to rerender the map. Nothing more.    Add this event outside this function on page load.

Reply:
I tried to implement it this way but I need the canvas for the Listeners. The canvas is only available once the map is composed.